Marlin/src/HAL/STM32/tft/tft_spi.cpp

```diff
@@ -34,35 +34,25 @@ DMA_HandleTypeDef TFT_SPI::DMAtx;
 void TFT_SPI::Init() {
   SPI_TypeDef *spiInstance;
 
-  #if PIN_EXISTS(TFT_RESET)
-    OUT_WRITE(TFT_RESET_PIN, HIGH);
-    HAL_Delay(100);
-  #endif
-
-  #if PIN_EXISTS(TFT_BACKLIGHT)
-    OUT_WRITE(TFT_BACKLIGHT_PIN, HIGH);
-  #endif
-
   OUT_WRITE(TFT_A0_PIN, HIGH);
   OUT_WRITE(TFT_CS_PIN, HIGH);
 
   if ((spiInstance = (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_SCK_PIN),  PinMap_SPI_SCLK)) == NP) return;
   if (spiInstance != (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_MOSI_PIN), PinMap_SPI_MOSI)) return;
 
-  #if PIN_EXISTS(TFT_MISO) && (TFT_MISO_PIN != TFT_MOSI_PIN)
-    if (spiInstance != (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_MISO_PIN), PinMap_SPI_MISO)) return;
+  #if PIN_EXISTS(TFT_MISO)
+    if (TFT_MISO_PIN != TFT_MOSI_PIN)
```

Comment:
Why did this change from compile-time check to runtime check?

Reply:
Also applies to additional checks later in the file.

Reply:
because pins (Pax, PBx) in F4 are `static uint_8`, they aren't macro... so we can use compile time `#if` for compare pins on F4

Reply:
When is a hat not a hat?
```cpp
TFT_MISO_PIN != TFT_MOSI_PIN
```

Reply:
Mosi master send data to slave, miso slave send data to master. When they are different, we have full duplex comm, if they are equal, we have only 1 line of data transfer, so we need to handle send and receive in the same pin.

Reply:
> we need to handle send and receive in the same pin.

Do you have an example of this. That simple sounds like "not SPI" to me.

I know some devices will have MOSI but no MISO. I am not aware of any mechanism to perform half-duplex I/O with SPI over one pin.

Reply:
Well... that is the original code from jmz... I just ported it to F4, keeping same logic. He must have his reasons for that....

I know that stm32duino needs to know and have a lot of checks for spi 1 line or spi 2 lines. There's some specific flags that need to be cleared in one or in the other case.

But I really don't have any TFT that don't have both miso/mosi. Maybe some generic spi TFT have only one these PINs...

That 2 line vs 1 line, is only handled here. The other hals don't do it... 

For me, I would keep only 2 lines support... but as the code is there, i don't think it hurts.

Reply:
Since it was already there, I think we can leave it here. It is possible that the STM32 SPI controllers support tri-stating the MOSI lines so the line could be shared, but more research would be needed to confirm or deny that.

I tend to think it is not useful unless we can find devices that we know use it. We could consider removing it in a future update.